File: api/internal/domain/branch/branch.go
package branch

import (
	"context"
	"errors"
	"time"

	"github.com/google/uuid"
)

var ErrBranchNotFound = errors.New("branch not found")

type Branch struct {
	ID           uuid.UUID
	Name         string
	City         string
	Address      string
	Region       string
	ContactName  string
	ContactPhone string
	Status       string // "active" | "inactive"
	PartnerID    *uuid.UUID
	PartnerName  string
	IsActive     bool
	CreatedAt    time.Time
	UpdatedAt    time.Time
}

type WriteRepository interface {
	Save(ctx context.Context, b *Branch) error
	Update(ctx context.Context, b *Branch) error
}

type ReadRepository interface {
	GetByID(ctx context.Context, id uuid.UUID) (*Branch, error)
	List(ctx context.Context, offset, limit int) ([]*Branch, int, error)
}

File: frontend/src/types/branch.ts
export interface Branch {
  id: string
  code: string
  name: string
  address: string
  city: string
  province: string
  phone: string
  email: string
  is_active: boolean
  created_at: string
  updated_at: string
}

export interface BranchFilters {
  is_active?: boolean
  page?: number
  limit?: number
}

export interface PaginatedBranches {
  data: Branch[]
  total: number
  offset: number
  limit: number
}
